### PR TITLE
Update the tracker used with batch execution

### PR DESCRIPTION
**Pull request by antalszava**

### CODE REVIEW

File: tests/test_integration.py [hunk 1]
```diff
@@ -569,3 +569,25 @@ def circuit(x, y):
         # Check that run was called twice: for the partial derivatives and for
         # running the circuit
         assert spy2.call_count == 2
+
+    def test_tracker(self):
+        """Tests the device tracker with batch execution."""
+        dev = qml.device('qiskit.aer', shots=100, wires=3)
+        x = np.array([0.1, 0.2])
```
Comment (rmoyard) [inline tests/test_integration.py:576]:
```suggestion
```

I think x is defined later.

**Review by rmoyard — COMMENTED** [review]
Thanks @antalszava, is that correct that you execute the circuit once for evaluating it and twice for the parameter shift rules?

**Comment by antalszava** [comment]
> is that correct that you execute the circuit once for evaluating it and twice for the parameter shift rules?

Yep. :slightly_smiling_face:

**Review by rmoyard — APPROVED** [review]
Thank you @antalszava 💯

**Pull request merged**